spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/AwsS3EnvironmentRepository.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.config.server.environment;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectIdBuilder;
+import org.yaml.snakeyaml.Yaml;
+
+import org.springframework.cloud.config.environment.Environment;
+import org.springframework.cloud.config.environment.PropertySource;
+import org.springframework.cloud.config.server.config.ConfigServerProperties;
+import org.springframework.core.Ordered;
+import org.springframework.util.StringUtils;
+
+/**
+ * @author Clay McCoy
+ */
+public class AwsS3EnvironmentRepository implements EnvironmentRepository, Ordered {
```

Comment:
To support the files API, this will need to implement `SearchPathLocator`. This is not required though.

Reply:
While not required, enabling the files/resources API would be desirable for this env repo. Does that require that the files be pre-fetched locally as happens with e.g. git clone, or can files be downloaded on demand?

Reply:
On demand is fine. `SearchPathLocator` implementation could happen in a follow on PR.

Reply:
Okay, not sure what Strings are expected to be put in Locations.locations. Any documentation on SearchPathLocator expectations?

Reply:
user can request an arbitrary path. In git any level of sub-directories.

Reply:
Right, so SearchPathLocator be somewhat moot when only one level is allowed? I guess it would return all files in that directory?